packages/compass-sidebar/src/components/multiple-connections/sidebar.tsx

```diff
@@ -78,15 +83,45 @@ function ConnectionErrorToastBody({
   );
 }
 
+async function copyConnectionString(connectionString: string) {
+  try {
+    await navigator.clipboard.writeText(connectionString);
+    openToast('copy-to-clipboard', {
+      title: 'Success',
+      description: 'Copied to clipboard.',
+      variant: 'success',
+      timeout: TOAST_TIMEOUT_MS,
+    });
+  } catch (err) {
+    openToast('copy-to-clipboard', {
+      title: 'Error',
+      description:
+        'An error occurred when copying to clipboard. Please try again.',
+      variant: 'warning',
+      timeout: TOAST_TIMEOUT_MS,
+    });
+  }
+}
+
 export function MultipleConnectionSidebar({
   activeWorkspace,
   onSidebarAction,
 }: MultipleConnectionSidebarProps) {
   const { openToast, closeToast } = useToast('multiple-connection-status');
   const cancelCurrentConnectionRef = useRef<(id: string) => Promise<void>>();
   const activeConnections = useActiveConnections();
+  const maybeProtectConnectionString = useMaybeProtectConnectionString();
 
   const [isConnectionFormOpen, setIsConnectionFormOpen] = useState(false);
+  const [isConnectionInfoModalOpen, setIsConnectionInfoModalOpen] = useState<
+    string | undefined
+  >();
```

Comment:
If `isConnectionInfoModalOpen` is a string, should this maybe be named differently to reflect that? If you want to keep the naming consistent, you can still use `isConnectionInfoModalOpen`, just maybe as inferred state in that case?

```suggestion
  const [connectionModalConnectionId, setConnectionModalConnectionId] = useState<
    string | undefined
  >();
  const isConnectionInfoModalOpen = !!connectionModalConnectionId;
```

(`connectionModalConnectionId` is just an example, there might be a better name for it 🙂 )

Reply:
you're right, I will rename it

Reply:
I did try this as a fixup, but the code was actually touched by several commits so I had conflicts and gave up.. another time 😄